client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx

```diff
@@ -67,48 +71,91 @@ const PendingGappsTosNotice = React.createClass( {
 
 		switch ( severity ) {
 			case 'warning':
-				return this.translate( 'Attention!', translationOptions );
+				return this.props.translate( 'Attention!', translationOptions );
 
 			case 'error':
-				return this.translate( 'Urgent!', translationOptions );
+				return this.props.translate( 'Urgent!', translationOptions );
 
 			default:
-				return this.translate( 'You\'re almost there!', translationOptions );
+				return this.props.translate( 'You\'re almost there!', translationOptions );
 		}
 	},
 
 	generateLogInClickHandler( { domainName, user, severity, isMultipleDomains } ) {
 		return () => {
-			this.recordEvent( 'pendingAccountLogInClick', { siteSlug: this.props.siteSlug, domainName, user, severity, isMultipleDomains, section: this.props.section } );
+			this.recordEvent( 'pendingAccountLogInClick', {
+				domainName,
+				isMultipleDomains,
+				severity,
+				user,
+				section: this.props.section,
+				siteSlug: this.props.siteSlug
+			} );
+		};
+	},
+
+	generateFixClickHandler() {
+		return () => {
+			this.recordEvent( 'fixPendingEmailSiteNoticeClick', this.props.siteSlug );
 		};
 	},
 
+	compactNotice() {
+		const severity = this.getNoticeSeverity(),
+			href = this.props.domains.length === 1
+				? domainManagementEmail( this.props.siteSlug, this.props.domains[ 0 ].name )
+				: domainManagementEmail( this.props.siteSlug );
+
+		return (
+			<Notice
+				isCompact={ this.props.isCompact }
+				status={ `is-${ severity }` }
+				showDismiss={ false }
+				key="pending-gapps-tos-acceptance-domain-compact"
+				text={ this.props.translate(
+					'Email requires action',
+					'Emails require action',
+					{
+						count: this.props.domains.length
+					}
+				) }>
+				<NoticeAction
+					href={ href }
+					onClick={ this.generateFixClickHandler() }>
+						{ this.props.translate( 'Fix' ) }
+				</NoticeAction>
+			</Notice>
+		);
+	},
+
 	oneDomainNotice() {
 		const severity = this.getNoticeSeverity(),
 			exclamation = this.getExclamation( severity ),
-			domainName = this.props.domains[0].name,
-			users = this.props.domains[0].googleAppsSubscription.pendingUsers;
+			domainName = this.props.domains[ 0 ].name,
```

Comment:
I agree, but our eslint rules throw a warning for it - it's the [`computed-property-spacing`](https://github.com/eslint/eslint/blob/master/docs/rules/computed-property-spacing.md) rule in our [`.eslintrc.js`](https://github.com/Automattic/wp-calypso/blob/master/.eslintrc.js#L24).